### DISCUSSION TITLE
Realtime Database and Functions?

### DISCUSSION THREAD

**Discussion by jonathan-chin** [capawesome-team/capacitor-firebase#568] [Q&A] [answered]
hi all,

I've got a project built with reactfire that I want to also support on mobile builds, through Capacitor and Ionic. It seems that there's support here for a lot of products like auth, firestore, and messaging.

I've noticed that Database and Functions are missing. Is there a particular reason why? do these need to use the native firebase sdk or can they use web?

**Comment by robingenz** [comment, marked as answer]
> I've noticed that Database and Functions are missing. Is there a particular reason why?

There is no specific reason. I just haven't gotten around to creating plugins for these two SDKs yet. 

> do these need to use the native firebase sdk or can they use web?

It depends on your use case whether you need these plugins or not. You can also create Capacitor apps using only the Firebase JS SDK. In some cases, however, it also makes sense to use the native SDKs.

**Reply by jonathan-chin** [reply]
thanks for your quick reply!

> You can also create Capacitor apps using only the Firebase JS SDK

do you know of any (even basic) examples? I've got 2 projects that have really close deadlines that might be faster to convert to plain JS.

> I just haven't gotten around to creating plugins for these two SDKs yet.

in general, I'd be happy to help and contribute, if that would be useful. but I'm in crunch time right now.

**Reply by robingenz** [reply]
> do you know of any (even basic) examples?

No, I currently have only a [demo app](https://github.com/robingenz/capacitor-firebase-plugin-demo) with the plugins.

**Reply by jonathan-chin** [reply]
hmm. would you welcome a `PR` to provide, at least initial, support for Database and Functions? I can use your existing files as guidelines for conventions and styles.

**Reply by robingenz** [reply]
Basically yes, but the effort required to create a plugin is relatively high. After all, you have to implement each plugin on all three platforms (Android, iOS, Web) and the code quality must be at least as high as with the existing plugins. It is best to use the Firebase Cloud Firestore plugin as an example, as it is the most up-to-date.

If you still want to work on the plugins, I can create the boilerplate code for you. But that will take around 2 days.

**Reply by jonathan-chin** [reply]
my thought would be to implement it incrementally, providing partial support and increasing feature support over time.

my projects need these products and I'm going to have to figure out a solution regardless.

let me do a small scale test using the firebase JS sdk in capacitor first and get back to you? I think that might resolve the short term issues.

also, thanks for all of your hard work on this. I empathize with what it's like to be a solo dev / small team on such a huge project.

**Reply by jonathan-chin** [reply]
thanks for your help and patience, @robingenz . I tested running the firebase js sdk on capacitor and everything works.

I know it's outside the scope of your project and goals, but I wonder why `reactfire` is having issues if it's just running the js sdk also. I'll ask over there. cheers!

**Reply by robingenz** [reply]
I'm glad that you have found a solution. I don't know what's the problem with reactfire. I've not tested it yet.
